File: java/src/useri/utils/CloneColumnPrefix.java
package useri.utils;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

import core.utils.Helper;
import core.utils.Perspectives;

/**
 * Prefix of columns within an *sps.cbs files, that represent individual clones.
 * @author noemi
 *
 */
public final class CloneColumnPrefix {
	private final static String[] VALUES=new String[] {"SP", "SP", "SP", "Clone","Clone"};
	private final static Perspectives[] KEYS=new Perspectives[]{Perspectives.ExomePerspective,Perspectives.GenomePerspective,Perspectives.KaryotypePerspective,Perspectives.TranscriptomePerspective,Perspectives.Identity};
	
	public static String getValue(Perspectives which) {
		return(VALUES[Helper.firstIndexOf(which, KEYS)]);
	}

	public static String[] values() {
		Set<String> temp = new HashSet<String>(Arrays.asList(VALUES));
		String[] uq = temp.toArray(new String[temp.size()]);
		return(uq);
	}
	
	
	
	
}

File: webportal/frontend/src/lib/cloneid/useri/CloneColumnPrefix.js

import { Perspectives } from '../core/utils/Perspectives';
import { Helper } from '../core/utils/Helper';

/**
 * Prefix of columns within an *sps.cbs files, that represent individual clones.
 */
export class CloneColumnPrefix {
    /**
     * @param {Perspectives} which
     */
    static getValue(which) {
        const index = Helper.firstIndexOf(which.name(), CloneColumnPrefix.KEYS);
        return CloneColumnPrefix.VALUES[index];
    }

    static values() {
        const uniqueValues = new Set(CloneColumnPrefix.VALUES);
        return Array.from(uniqueValues);
    }

    static VALUES = ["SP", "SP", "SP", "Clone","SP","Clone"];
    static KEYS = [
        Perspectives.ExomePerspective,
        Perspectives.GenomePerspective,
        Perspectives.KaryotypePerspective,
        Perspectives.TranscriptomePerspective,
        Perspectives.MorphologyPerspective,
        Perspectives.Identity
      ];
}
